File: CTCI_1_String_Array/ctci_1_3.cpp
#include <iostream>
#include <cstring>
using namespace std;

void removeDuplicate(char *s){
	int len = strlen(s);
	if(len < 2)
		return;
	bool a[256];
	for(int i=0; i<256; i++){
		a[i] = true;
	}
	int k = 0;
	for(int i=0; i<len; i++){
		if(a[s[i]]){
			s[k] = s[i];
			k++;
			a[s[i]] = false;
		}
	}
	s[k] = '\0';
}

int main(){
	char s1[] = "qwerty";
	char s2[] = "qqqqqq";
	char s3[] = "";
	char s4[] = "qweqwt";
	removeDuplicate(s1);
	removeDuplicate(s2);
	removeDuplicate(s3);
	removeDuplicate(s4);
	cout<<s1<<" "<<s2<<" "<<s3<<" "<<s4<<endl;
	return 0;
}

File: CTCI_1_String_Array/ctci_1_3.py
def removeDuplicate(s):
	ans = []
	for i in s:
		if i not in ans:
			ans.append(i)
	return ''.join(ans)

def main():
	s1 = 'qwerty'
	s2 = 'qqqqqq'
	s3 = ''
	s4 = 'qweqwt'
	print removeDuplicate(s1)
	print removeDuplicate(s2)
	print removeDuplicate(s3)
	print removeDuplicate(s4)

if __name__ == '__main__':
	main()